Source/WebCore/page/IntersectionObserver.cpp

```diff
@@ -143,9 +143,13 @@ IntersectionObserver::IntersectionObserver(Document& document, Ref<IntersectionO
     } else if (root) {
         auto& observerData = downcast<Element>(*root).ensureIntersectionObserverData();
         observerData.observers.append(*this);
-    } else if (auto* frame = document.frame()) {
-        if (auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame()))
-            m_implicitRootDocument = localFrame->document();
+    } else {
+        SUPPRESS_UNCOUNTED_LOCAL auto* frame = document.frame();
+        if (frame) {
+            SUPPRESS_UNCOUNTED_LOCAL auto* localFrame = dynamicDowncast<LocalFrame>(frame->mainFrame());
```

Comment:
These two look like a bug in static analyzer. Everything in this code block is trivial so we shouldn't be needing to suppress the warning.

Reply:
I removed this part.